### ISSUE TITLE
invalid ID attribute and character stripping

### ISSUE THREAD

**Issue by bartoleo** [hakimel/reveal.js#1230]
I'm using a tool to create reveal.js slides from asciidoctor files.
Asciidoctor creates sections with id containing characters outside (a-z,A-Z,0-9,.,-,_,:), e.g.:
à è ì ò ù ...

readUrl functions checks id:

```
// Ensure the named link is a valid HTML ID attribute
            if( /^[a-zA-Z][\w:.-]*$/.test( name ) ) {
```

writeUrl strips charactees:

```
id = id.replace( /[^a-zA-Z0-9\-\_\:\.]/g, '' );
```

These sections are not linkable from url.

There's a reason for these 'checks'?
HTML5 is very permissive about id attribute:
http://www.w3.org/TR/html5/dom.html#the-id-attribute

**Comment by hakimel** [comment]
Happy to make this more flexible than the current HTML 4 validation.

**Comment by Peque** [comment]
[Another consequence of this](1346): if you execute `grunt serve` and modify the HTML file while displaying the slide with a "broken" ID in the browser (imagine a live preview of your presentation while editing it), it will automatically jump to the first slide, instead of displaying the same slide after the edit. Which means after each edit you will have to navigate again to that slide to see the changes.

**Comment by hakimel** [comment]
This was fixed in #2037. The fix is available from the dev branch until next release.
